server/auth.go

```diff
@@ -525,6 +525,7 @@ func (s *Server) processClientOrLeafAuthentication(c *client, opts *Options) boo
 			c.Debugf("Account JWT not signed by trusted operator")
 			return false
 		}
+		// this only executes IF there's an issuer on the Juc - otherwise the account is already
 		if juc.IssuerAccount != "" && !acc.hasIssuer(juc.Issuer) {
```

Comment:
why go back to this from what we ended up with?

Reply:
because in the end the function that we dispatch is doing that check. Here we are propagating knowledge of an edge condition where the issuer account is set, but it also matches the account name. Let the `hasIssuer()` verification do what it does.

Reply:
ok